### PR TITLE
Add more testing and restrictions to patient's fields

### PR DESCRIPTION
**Pull request by PeiYee88**
Add test-cases to
1. `Parsers` for `get /hw` `get /nok` `get /appton`
2. `Patient's field Predicate` for `hospital wing` `appointment by date predicate`

Restricts hospital wing to only accept values of {east, west, south, north; case-insensitive} only

Reconsider implementation of AppointmentByDatePredicate to include cases where past appointments is null

`get /nok` accepts multiple inputs
`get /hw` accepts only one input
`get /appton` accepts multiple inputs : which allows medical staffs to be able to retrieve multiple appointment dates in a single command

### CODE REVIEW

**Review by nehcuy — CHANGES_REQUESTED** [review]
I think there are better ways to handle hospital wing restrictions. It would be better if we rethink the logic through the processing of user inputs. :) I've wrote up the possible changes to make the implementation seem more consistent with the rest of the codebase.

File: src/main/java/seedu/address/model/person/HospitalWing.java [hunk 1]
```diff
@@ -3,18 +3,51 @@
...
+            public String toString() {
+                return "east";
+            }
+        },
+        WEST {
+            @Override
+            public String toString() {
+                return "west";
+            }
+        },
```
[the remark this answers is not in the document]
Comment (PeiYee88) [reply src/main/java/seedu/address/model/person/HospitalWing.java:48]: no need semicolon

File: src/main/java/seedu/address/model/person/AppointmentByDatePredicate.java [hunk 2]
```diff
@@ -25,18 +25,31 @@ public AppointmentByDatePredicate(List<LocalDate> appointments) {
...
+
+    private boolean isPresentUpcomingAppointment(Person person) {
+        return !appointments.stream().anyMatch(keyword
+                -> person.getUpcomingAppointment().get().value == null);
+    }
+
+    private boolean isPresentPastAppointment(Person person) {
+        return appointments.stream().anyMatch(keyword
+                -> person.getPastAppointments().size() != 0);
     }
```
[the remark this answers is not in the document]
Comment (PeiYee88) [reply src/main/java/seedu/address/model/person/AppointmentByDatePredicate.java:53]: edited! thanks for spoting it out!

File: src/main/java/seedu/address/model/person/HospitalWing.java [hunk 3]
```diff
@@ -30,7 +63,8 @@ public HospitalWing(String hospitalWing) {
      * Returns true if a given string is a valid hospital wing.
      */
     public static boolean isValidHospitalWing(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return Arrays.stream(HospitalWingTypes.values()).anyMatch(hospitalwingtype ->
+                test.toLowerCase().equals(hospitalwingtype.toString()));
     }
```
[the remark this answers is not in the document]
Comment (nehcuy) [reply src/main/java/seedu/address/model/person/HospitalWing.java]: Apologies, there seems to be some errors caused by this. Fixed in PR #116.

**Pull request merged**